### PR TITLE
Adds geo and install plugin control to extensions featured page

### PR DESCRIPTION
**Pull request by belcherj**
On the featured tab of the Extensions page we show extensions.

- This PR hides extensions that are activated on the site.
- This PR also adds the concept of geo blacklist and whitelists.

The featured.json version bumped to 2 because adding more than three extensions per section will show all of them in previous versions.  We need to add more than three extensions for when extensions are not shown due to the above constraints.

In order to test:

- Go to featured tab of the extensions page
- Observe that any extensions that you have activated on that site are not shown

To test the geo gating you will need to download the featured-v2.json file and add:

"geowhitelist": [ "GB" ] or "geoblacklist": [ "US" ] to one or more of the extensions.  Do not add both to one since that wouldn't make any sense. Once you have the modified file put it somewhere web accessible and change the url on line 27 of includes/admin/class-wc-admin-addons.php.

A whitelist is nothing but the countries in the list.
A blacklist is everything but the countries in the list.

### CODE REVIEW

**Comment by claudiulodro** [comment]
The code for this looks good, but the json file is not decoding properly. `json_decode( wp_remote_retrieve_body( $raw_featured ) )` is returning `null`. When I run the contents of the json file through [this online decoder](http://freeonlinetools24.com/json-decode) I get an "Invalid JSON format!!" error.

**Comment by belcherj** [comment]
Fixed, there were some extra commas.

File: includes/admin/class-wc-admin-addons.php [hunk 1]
```diff
@@ -506,4 +508,29 @@ public static function install_woocommerce_services_addon() {
...
+	 * Should an extension be shown on the featured page.
+	 *
+	 * @param object $item
+	 *
+	 * @return boolean
+	 */
+	public static function show_extension( $item ) {
+		$location = WC()->countries->get_base_country();
+		$show = true;
+		if ( isset( $item->geowhitelist ) && ! in_array( $location, $item->geowhitelist ) ) {
```
Comment (mikejolley) [inline includes/admin/class-wc-admin-addons.php]: Add strict true (3rd param) to satisfy PHPCS.

File: includes/admin/class-wc-admin-addons.php [hunk 2]
```diff
@@ -506,4 +508,29 @@ public static function install_woocommerce_services_addon() {
...
+			$show = false;
+		}
+
+		if ( isset( $item->geoblacklist ) && in_array( $location, $item->geoblacklist ) ) {
```
Comment (mikejolley) [inline includes/admin/class-wc-admin-addons.php]: Add strict true (3rd param) to satisfy PHPCS.

File: includes/admin/class-wc-admin-addons.php [hunk 3]
```diff
@@ -506,4 +508,29 @@ public static function install_woocommerce_services_addon() {
...
+	 *
+	 * @param object $item
+	 *
+	 * @return boolean
+	 */
+	public static function show_extension( $item ) {
+		$location = WC()->countries->get_base_country();
+		$show = true;
+		if ( isset( $item->geowhitelist ) && ! in_array( $location, $item->geowhitelist ) ) {
+			$show = false;
```
Comment (mikejolley) [inline includes/admin/class-wc-admin-addons.php]: If these conditions all return false, why not return here and dump the `$show` variable? If it gets to the end of the func, return true.

File: includes/admin/class-wc-admin-addons.php [hunk 4]
```diff
@@ -506,4 +508,29 @@ public static function install_woocommerce_services_addon() {
 		wp_safe_redirect( remove_query_arg( array( 'install-addon', '_wpnonce' ) ) );
 		exit;
 	}
+
+	/**
+	 * Should an extension be shown on the featured page.
+	 *
+	 * @param object $item
+	 *
```
Comment (mikejolley) [inline includes/admin/class-wc-admin-addons.php]: I don't think white space is needed here.

**Review by mikejolley — APPROVED** [review]
Just some nitpicks then it can merge.

**Pull request merged**